app/src/main/java/com/rootstrap/android/ui/activity/main/SignUpActivityViewModel.kt

```diff
@@ -23,26 +26,29 @@ open class SignUpActivityViewModel(listener: ViewModelListener?) : BaseViewModel
 
     fun signUp(user: User) {
         networkState = NetworkState.loading
-        manager.signUp(user)
-    }
+        viewModelScope.launch {
+            val result = manager.signUp(user = user)
 
-    @Subscribe
-    fun signedUpSuccessfully(event: UserManager.UserCreatedSuccessfullyEvent) {
-        networkState = NetworkState.idle
-        state = SignUpState.signedUpSuccess
-    }
+            if (result.isSuccess) {
+                result.getOrNull()?.value?.user?.let { user ->
+                    SessionManager.signIn(user)
+                }
 
-    @Subscribe
-    fun signedUpError(event: ErrorEvent) {
-        error = event.error
-        networkState = NetworkState.idle
-        networkState = NetworkState.error
+                networkState = NetworkState.idle
+                state = SignUpState.signedUpSuccess
```

Comment:
I know this is old, but it should be either `signUpSuccess` or `signedUpSuccessfully`.